Meta/serenity_gdb.py

```diff
@@ -0,0 +1,263 @@
+
+        inner_type_ptr = self.val.type.template_argument(0).pointer()
+
+        if int(outline_buf) != 0:
+            elements = outline_buf.cast(inner_type_ptr)
+        else:
+            elements = get_field_unalloced(self.val, "m_inline_buffer_storage", inner_type_ptr)
+
+        return [("[%d]" % (i,), elements[i]) for i in range(vec_len)]
+
+
+class AKHashMapPrettyPrinter(object):
+    def __init__(self, val):
+        self.val = val
+
+    @staticmethod
+    def _iter_hashtable(val, cb):
+        entry_type_ptr = val.type.template_argument(0).pointer()
+        buckets = val["m_buckets"]
+        for i in range(0, val["m_capacity"]):
+            bucket = buckets[i]
+            if bucket["used"]:
+                cb(bucket["storage"].cast(entry_type_ptr))
+
+    @staticmethod
+    def _iter_hashmap(val, cb):
+        table = val["m_table"]
+        AKHashMapPrettyPrinter._iter_hashtable(table, lambda entry: cb(entry["key"], entry["value"]))
+
+    def to_string(self):
+        return self.val.type.name
+
+    def children(self):
+        elements = []
+
+        def cb(key, value):
+            nonlocal elements
+            elements.append(("[%s]" % (str(key),), value))
+
+        AKHashMapPrettyPrinter._iter_hashmap(self.val, cb)
+        return elements
+
+
+class VirtualAddress(object):
+    def __init__(self, val):
+        self.val = val
+
+    def to_string(self):
+        return self.val["m_address"]
+
+
+class SerenityPrettyPrinterLocator(gdb.printing.PrettyPrinter):
+    def __init__(self):
+        super(SerenityPrettyPrinterLocator, self).__init__("serenity_pretty_printers", [])
+
+        self.re = re.compile('^([^<]+)(<.*>)?$')
+
+    def __call__(self, val):
+        type = gdb.types.get_basic_type(val.type)
+        typename = str(type.tag)
+
+        match = self.re.match(typename)
+        if not match:
+            return None
+
+        klass = match.group(1)
+
+        if klass == 'AK::Atomic':
+            return AKAtomic(val)
+        elif klass == 'AK::DistinctNumeric':
+            return AKDistinctNumeric(val)
+        elif klass == 'AK::HashMap':
+            return AKHashMapPrettyPrinter(val)
+        elif klass == 'AK::RefCounted':
+            return AKRefCounted(val)
+        elif klass == 'AK::RefPtr':
+            return AKRefPtr(val)
+        elif klass == 'AK::OwnPtr':
+            return AKOwnPtr(val)
+        elif klass == 'AK::NonnullRefPtr':
+            return AKRefPtr(val)
+        elif klass == 'AK::String':
+            return AKString(val)
+        elif klass == 'AK::StringView':
+            return AKStringView(val)
+        elif klass == 'AK::StringImpl':
+            return AKStringImpl(val)
+        elif klass == 'AK::Vector':
+            return AKVector(val)
+        elif klass == 'VirtualAddress':
+            return VirtualAddress(val)
+
+
+gdb.printing.register_pretty_printer(None, SerenityPrettyPrinterLocator(), replace=True)
+
+
+class FindThreadCmd(gdb.Command):
```

Comment:
Can you write python help text, so it will show up when someone asks for help of this command in gdb? 

Reference: https://sourceware.org/gdb/onlinedocs/gdb/Commands-In-Python.html

> The help text for the new command is taken from the Python documentation string for the command’s class, if there is one. If no documentation string is provided, the default value “This command is not documented.” is used.